File: hw/maisincho/u3/HW20Compiste/CompositePy/model/President.py
from model.Supervisor import Supervisor  

class President(Supervisor):
    _president_instance = None  # instance

    def __new__(cls, *args, **kwargs):
        #only instance
        if cls._president_instance is None:
            cls._president_instance = super().__new__(cls)
        return cls._president_instance

    def __init__(self, name=None):
        if not hasattr(self, '_initialized'):
            super().__init__()
            self.title = "President"
            self.name = name if name else ""
            self._initialized = True  # to avoid initializing on new calls

    def state_name(self):
        super().state_name()

    @classmethod
    def get_president(cls, name):
        instance = cls(name)
        instance.name = name
        return instance

File: hw/palacios/u3/HW20Composite/CompositePattern-js/model/President.js
const Supervisor = require('./Supervisor');

class President extends Supervisor {
    static #instance = null;
    
    constructor(aName) {
        if (President.#instance && !aName) {
            return President.#instance;
        }
        
        super();
        if (aName) this.name = aName;
        this.title = "President";
        
        if (!President.#instance) {
            President.#instance = this;
        }
        return President.#instance;
    }
    
    static getPresident(aName) {
        if (!President.#instance) {
            President.#instance = new President(aName);
        } else if (aName) {
            President.#instance.name = aName;
        }
        return President.#instance;
    }
}

module.exports = President;